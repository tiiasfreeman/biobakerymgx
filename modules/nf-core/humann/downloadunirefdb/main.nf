process HUMANN_DOWNLOADUNIREFDB {
    tag "${params.uniref_db_version}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/humann:3.8--pyh7cba7a3_0':
        'quay.io/biocontainers/humann:3.8--pyh7cba7a3_0' }"

    output:
    path "${prefix}/uniref/"    , emit: uniref_db_dir
    path "versions.yml"         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "uniref_db_${params.uniref_db_version}"
    """
    mkdir -p ${prefix}

    https_proxy=http://klone-dip1-A-ib:3128
    export https_proxy
    humann_databases \\
        --download uniref ${params.uniref_db_version} \\
        ${prefix} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        humann: \$(echo \$(humann --version 2>&1 | sed 's/^.*humann //; s/Using.*\$//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "uniref_db_${params.uniref_db_version}"
    """
    mkdir -p ${prefix}/uniref/

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        humann: \$(echo \$(humann --version 2>&1 | sed 's/^.*humann //; s/Using.*\$//' ))
    END_VERSIONS
    """
}
