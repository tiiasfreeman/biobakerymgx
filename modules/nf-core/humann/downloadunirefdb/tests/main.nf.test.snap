{
    " uniref_db_version = 'DEMO_diamond' ": {
        "content": [
            {
                "0": [
                    [
                        "uniref90_demo_prots_v201901b.dmnd:md5,1d2f8df48a349d72436425aafa29a6b1"
                    ]
                ],
                "1": [
                    "versions.yml:md5,9701558bc32dc67749aad70c9a67960f"
                ],
                "uniref_db_dir": [
                    [
                        "uniref90_demo_prots_v201901b.dmnd:md5,1d2f8df48a349d72436425aafa29a6b1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9701558bc32dc67749aad70c9a67960f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-16T14:04:05.712218097"
    },
    " uniref_db_version = 'DEMO_diamond' -stub ": {
        "content": [
            {
                "0": [
                    [
                        
                    ]
                ],
                "1": [
                    "versions.yml:md5,9701558bc32dc67749aad70c9a67960f"
                ],
                "uniref_db_dir": [
                    [
                        
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9701558bc32dc67749aad70c9a67960f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-16T14:04:13.931593633"
    }
}